src/Pages/Signup.js

```diff
@@ -0,0 +1,116 @@
+import React,{useReducer,useState,useEffect} from 'react'
+import axios from 'axios'
+import {Link,useNavigate} from 'react-router-dom'
+
+export function Signup() {
+    const navigate = useNavigate()
+
+    const [textInputState,textInputDispatcher] = useReducer(inputDispatcher,{username:"",password:"",rePassword:"",answer:""})
+    
+    const [visibilityErr,setErrVisibility] = useState("hidden")
```

Comment:
Since you're already using `reducer` in this same file, you can move these state variables to reducer too!